### ISSUE TITLE
Panel is not fully synchronized with dconf /org/cinnamon/enabled-applets

### ISSUE THREAD

**Issue by alex-ie** [linuxmint/cinnamon#10739]
* Cinnamon version 5.0.4
 * Distribution Mint 20.2
 * 64 bit
 
**Issue**
When I boot I get `['']` for output of `dconf read /org/cinnamon/enabled-applets` but Cinnamon panel still contains "usual" "standard" applets. 

**Steps to reproduce**
I boot LM with `dconf write /org/cinnamon/enabled-applets "['']"` run on systemd's `WantedBy=multi-user.target`. As a result after boot is complete in terminal I get `['']` for output of `dconf read /org/cinnamon/enabled-applets` but panel still has "usual" look filled with applets.

**Expected behaviour**
Panel looks in accordance with changed `dconf /org/cinnamon/enabled-applets`.

**Other information**
IMO it is a bug as output of dconf changes, but panel does not. As a workaround I would appreciate to know another way to tweak the standard panel during boot time. First I tried to add applets this way but failed now I've tried to write empty array. Writing empty array _after_ boot and login works in changing the panel, but I'd like to be able to tweak the panel during boot. Also it is interesting where Cinnamon takes info for the panel during boot if not from dconf. Thanks.

**Comment by mtwebster** [comment]
Dconf needs the session dbus running before changes can be written. Is it running after the dbus target is reached? Does `dconf update` help?

All I can think is for whatever reason Cinnamon is reading that key before dconf finishes syncing - this seems unlikely, though, but it may be grabbing the 'default' values, which I don't think are reliant on dbus.

I'm curious what `gsettings get org.cinnamon enabled-applets` returns vs `dconf read` (or `gsettings set` vs `dconf write`). I only ever use the dconf backend if I was to backup my settings to be able to reload later - I'm not sure what its update mechanics are.

If you restart cinnamon after the session loads, do the applets load again? (alt-f2, r, enter)

As for your desired result, you can use overrides and also lock down schemas from being changed.

There's a lot here:
https://help.gnome.org/admin/system-admin-guide/stable/setup.html.en

For a working example, on Mint, check out /usr/share/glib-2.0/schemas/mint-artwork.gschema.override.  Note this only overrides the defaults. I've never messed with locking settings.

If actually want to allow changes and then reset each time you log in, you might consider an autostart desktop file. While the session is starting (after you log in), there are phases the session manager goes thru while starting components. Some want/need to start before others.

Have a look at `/etc/xdg/autostart/cinnamon-settings-daemon-power.desktop` - Initialization phase is just before cinnamon is started. Typically programs in this phase are required to register with the session manager or else they'll hold up log-in, but if it's a quick single-liner it may be fine.

**Comment by alex-ie** [comment]
@mtwebster Thanks a lot for comprehensive response. I have not gone through all your recommendations, but I feel it is polite to give answers to the questions I can answer now. 

During several boots I've noted resulting panel depends on something and it is my guess it it the speed of boot process (hardware): in relatively slow boot from USB 2.0 directly panel got updated, but after faster boot (boot from RAM) has not.

I guess dbus is running, I also update e.g. Cinnamon sound scheme in same script, set custom keyboard shortcuts and sounds got updated fine, shortcuts work after login.

`gsetting get` resulted in same as `dconf read`, I have not tried `set` yet. Restarting Cinnamon resulted in panel update to dconf data (my target result), suspend/resume resulted in dconf updated to "default" values. Do you know where and how those "defaults" are stored?

**Comment by alex-ie** [comment]
@mtwebster, I've learned to use overrides for scheme, for my purpose I consider it is even more clean way than `dconf write` on boot. So thanks again and I'm closing the issue. Thanks again!

P.S. for future reference, I've done some more boots and what I got is:
When I did `dconf write /org/cinnamon/enabled-applets "['panel1:left:0:menu@cinnamon.org:0']"` +  `gsettings set org.cinnamon enabled-applets"['panel1:left:0:menu@cinnamon.org:0']"`, then I've got default panel both on fast and slower hardware boot (several times), when I restarted Cinnamon, panel was updated to menu only.
